hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionFileSystem.java

```diff
@@ -630,19 +630,26 @@ public Path commitDaughterRegion(final RegionInfo regionInfo)
   /**
    * Create the region splits directory.
    */
-  public void createSplitsDir() throws IOException {
+  public void createSplitsDir(RegionInfo daughterA, RegionInfo daughterB) throws IOException {
     Path splitdir = getSplitsDir();
     if (fs.exists(splitdir)) {
       LOG.info("The " + splitdir + " directory exists.  Hence deleting it to recreate it");
       if (!deleteDir(splitdir)) {
-        throw new IOException("Failed deletion of " + splitdir
-            + " before creating them again.");
+        throw new IOException("Failed deletion of " + splitdir + " before creating them again.");
       }
     }
     // splitDir doesn't exists now. No need to do an exists() call for it.
     if (!createDir(splitdir)) {
       throw new IOException("Failed create of " + splitdir);
     }
+    Path daughterATmpDir = getSplitsDir(daughterA);
```

Comment:
Why this change?

Reply:
We have to create the directories for daughter regions here, as it is possible that we do not have any  store files to split, which means the directories may not be created when we call `commitDaugherRegion`, and then we will miss the .regioninfo file and cause TestHbck to fail. Not sure why this does not have problem in the past but it is no harm...

Reply:
ok